php-checks/src/main/java/org/sonar/php/checks/ConstructorDeclarationCheck.java

```diff
@@ -72,4 +75,23 @@ public void visitNode(Tree tree) {
     }
   }
 
+  /**
+   * If there is a namespace that is declared at the file script level, then all classes belong to this namespace.
+   * If there are several namespaces, classes can belong to different namespaces,
+   * but each class must then belong to one namespace
+   */
+  private static boolean isClassInNamespaceContext(ClassDeclarationTree classDec) {
+    Tree parent = classDec.getParent();
+    if (parent != null && parent.is(Kind.SCRIPT)) {
+      ScriptTree scriptTree = (ScriptTree) parent;
+
+      for (Tree statement : scriptTree.statements()) {
+        if (statement.is(Kind.NAMESPACE_STATEMENT)) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
```

Comment:
This implementation still raises a FP on:
```
if (true) {
  class D {
    public function D() {   // OK
    }
  }
}
```
Another approach could be something like:
```
    Symbol symbol = context().symbolTable().getSymbol(classDec.name());
    QualifiedName qualifiedName = symbol.qualifiedName();
    return qualifiedName != null && !qualifiedName.toString().equals(symbol.name());
```